crypto/curve25519/curve25519.c

```diff
@@ -909,65 +849,65 @@ static void fe_pow22523(fe out, const fe z) {
   fe t2;
   int i;
 
-  fe_sq(t0, z);
-  fe_sq(t1, t0);
+  GFp_fe_sq(t0, z);
+  GFp_fe_sq(t1, t0);
   for (i = 1; i < 2; ++i) {
-    fe_sq(t1, t1);
+    GFp_fe_sq(t1, t1);
   }
-  fe_mul(t1, z, t1);
-  fe_mul(t0, t0, t1);
-  fe_sq(t0, t0);
-  fe_mul(t0, t1, t0);
-  fe_sq(t1, t0);
+  GFp_fe_mul(t1, z, t1);
+  GFp_fe_mul(t0, t0, t1);
+  GFp_fe_sq(t0, t0);
+  GFp_fe_mul(t0, t1, t0);
+  GFp_fe_sq(t1, t0);
   for (i = 1; i < 5; ++i) {
-    fe_sq(t1, t1);
+    GFp_fe_sq(t1, t1);
   }
-  fe_mul(t0, t1, t0);
-  fe_sq(t1, t0);
+  GFp_fe_mul(t0, t1, t0);
+  GFp_fe_sq(t1, t0);
   for (i = 1; i < 10; ++i) {
-    fe_sq(t1, t1);
+    GFp_fe_sq(t1, t1);
   }
-  fe_mul(t1, t1, t0);
-  fe_sq(t2, t1);
+  GFp_fe_mul(t1, t1, t0);
+  GFp_fe_sq(t2, t1);
   for (i = 1; i < 20; ++i) {
-    fe_sq(t2, t2);
+    GFp_fe_sq(t2, t2);
   }
-  fe_mul(t1, t2, t1);
-  fe_sq(t1, t1);
+  GFp_fe_mul(t1, t2, t1);
+  GFp_fe_sq(t1, t1);
   for (i = 1; i < 10; ++i) {
-    fe_sq(t1, t1);
+    GFp_fe_sq(t1, t1);
   }
-  fe_mul(t0, t1, t0);
-  fe_sq(t1, t0);
+  GFp_fe_mul(t0, t1, t0);
+  GFp_fe_sq(t1, t0);
   for (i = 1; i < 50; ++i) {
-    fe_sq(t1, t1);
+    GFp_fe_sq(t1, t1);
   }
-  fe_mul(t1, t1, t0);
-  fe_sq(t2, t1);
+  GFp_fe_mul(t1, t1, t0);
+  GFp_fe_sq(t2, t1);
   for (i = 1; i < 100; ++i) {
-    fe_sq(t2, t2);
+    GFp_fe_sq(t2, t2);
   }
-  fe_mul(t1, t2, t1);
-  fe_sq(t1, t1);
+  GFp_fe_mul(t1, t2, t1);
+  GFp_fe_sq(t1, t1);
   for (i = 1; i < 50; ++i) {
-    fe_sq(t1, t1);
+    GFp_fe_sq(t1, t1);
   }
-  fe_mul(t0, t1, t0);
-  fe_sq(t0, t0);
+  GFp_fe_mul(t0, t1, t0);
+  GFp_fe_sq(t0, t0);
   for (i = 1; i < 2; ++i) {
-    fe_sq(t0, t0);
+    GFp_fe_sq(t0, t0);
   }
-  fe_mul(out, t0, z);
+  GFp_fe_mul(out, t0, z);
 }
 
 void GFp_x25519_ge_tobytes(uint8_t *s, const ge_p2 *h) {
   fe recip;
   fe x;
   fe y;
 
-  fe_invert(recip, h->Z);
-  fe_mul(x, h->X, recip);
-  fe_mul(y, h->Y, recip);
+  eddsa_elem_invert(recip, h->Z);
```

Comment:
I suggest that, as the very first commit, we find a way to pull out the call to eddsa_elem_invert from C to Rust. e.g. we could add `recip` as a parameter to `GFp_x25519_ge_tobytes`. That way we can avoid the no_mangle and naming stuff.